mono/profiler/proflog.c

```diff
@@ -2490,6 +2541,517 @@ counters_and_perfcounters_sample (MonoProfiler *prof)
 	perfcounters_sample (prof, (now - start) / 1000/ 1000);
 }
 
+#define COVERAGE_DEBUG(x) if (debug_coverage) {x}
+static GHashTable *coverage_methods = NULL;
```

Comment:
I'm not entirely sure, but I *think* these can be made into concurrent hash tables (`mono/utils/mono-conc-hashtable.h`) to eliminate the mutex. That said, we'd need a way to iterate the concurrent hash tables in situations where we know no concurrent access happens (i.e. in `dump_coverage`). If we don't do this, I think we'd end up seeing huge contention in `method_enter` for apps with loads of methods.